python/sglang/srt/model_executor/model_runner.py

```diff
@@ -514,8 +517,22 @@ def update_weights_from_distributed(self, name, dtype, shape):
             logger.error(error_msg)
             return False, error_msg
 
-    def update_weights_from_tensor(self, named_tensors: List[Tuple[str, torch.Tensor]]):
-        self.model.load_weights(named_tensors)
+    def update_weights_from_tensor(
+        self,
+        named_tensors: List[Tuple[str, Union[torch.Tensor, "LocalSerializedTensor"]]],
+        load_format: Optional[str] = None,
+    ):
+        named_tensors = [
+            (name, _unwrap_tensor(tensor, tp_rank=self.tp_rank))
+            for name, tensor in named_tensors
+        ]
+        # TODO should we name it "direct" or "megatron"?
```

Comment:
based on its implementation, I recommend "direct".

Reply:
P.S. #2736 named it "megatron", while I feel "direct" may be a bit more suitable, thus I leave the question here.